pkg/api/addons.go

```diff
@@ -232,7 +240,7 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultAzureCNINetworkMonitorAddonsConfig := KubernetesAddon{
 		Name:    AzureCNINetworkMonitoringAddonName,
-		Enabled: azureCNINetworkMonitorAddonEnabled(o),
+		Enabled: to.BoolPtr(o.IsAzureCNI() && o.KubernetesConfig.NetworkPolicy != NetworkPolicyCalico),
```

Comment:
Carried this additional default gate over from the "last mile" "concatenate addons string" business logic area.